### ISSUE TITLE
Investigating: Weird build error when trying to cleanup imports

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/angular_components_example#70]
See this branch https://github.com/kevmoo/angular_components_example/tree/kevmoo.haxs

```console
$ clone above
$ cd example/angular_components_example/
$ pub get
$ pub run build_runner build --verbose

...

[FINE] sass_builder:read file: angular_components|lib/css/material/const/_typography.scss
[FINE] sass_builder:found imports:
[FINE] sass_builder:compiling file: /private/var/folders/9_/hw8ty6ns5c31y68b55m6v3bw0071_k/T/scratch_spaceGRPd8t/packages/material_dialog_example/material_dialog_example.scss
[FINE] sass_builder:wrote css file: lib/material_dialog_example.scss.css
[SEVERE] angular_gallery_section on package:material_select_example/material_dropdown_select_demo.dart:

AssetNotFoundException: material_select_example|lib/material_dropdown_select_demo.template.dart
[SEVERE] Build:
Failed after 3.2s
AssetNotFoundException: material_select_example|lib/material_dropdown_select_demo.template.dart
#0      ImportScanner._scanImportPaths (package:angular_gallery_section/import_scanner.dart:86:21)
<asynchronous suspension>
#1      ImportScanner.scanForComponent (package:angular_gallery_section/import_scanner.dart:64:26)
<asynchronous suspension>
#2      ComponentApiBuilder._findDocsMap (package:angular_gallery_section/builder/component_api_builder.dart:65:38)
<asynchronous suspension>
#3      ComponentApiBuilder._mustacheContext (package:angular_gallery_section/builder/component_api_builder.dart:102:17)
<asynchronous suspension>
#4      ComponentApiBuilder.build (package:angular_gallery_section/builder/component_api_builder.dart:38:35)
<asynchronous suspension>
```

**Comment by kevmoo** [comment]
Guessing this is *not* an issue w/ this repo. Just a placeholder until we figure out where the bug is...

**Comment by nshahan** [comment]
It could just be that the uber import is missing some parts of the library. It is hand written. If that is really going to be the only entry point it would need to be generated.

**Comment by natebosch** [comment]
Whatever the root cause I think we also have an issue with `build_runner` here - we aren't writing the asset graph for this build...

**Comment by nshahan** [comment]
The angular_gallery_section package has a builder that reads imports, loads the files as assets looking for documentation and examples. It is possible there are some assumptions being broken by the single entry point. This will probably significantly increase the time it takes to run the build if all files are now part of every search.

**Comment by natebosch** [comment]
Filed https://github.com/dart-lang/build/issues/1207
Will have a fix out for that soon which should make it easier to diagnose the root cause here since we'll be able to inspect an asset graph.

**Comment by natebosch** [comment]
The root cause here is an ordering issue.

`package:material_select_example/material_dropdown_select_demo.template.dart` is created in phase 368

One of the failing actions here is attempting to produce `package:material_select_example/material_dropdown_select_demo.api.dart` which is in phase 364 so it's not allowed to read an output from phase 368 but it tries to read that file

@nshahan - Currently `angular_gallery_section` has `runs_before: ["angular|angular"]` - do you know why it would need to read the `.template.dart` file? Is our ordering configuration insufficient here?

**Comment by nshahan** [comment]
Yeah I see why, it is listed as an import:
https://github.com/dart-lang/angular_components_example/blob/master/example/material_select_example/lib/material_dropdown_select_demo.dart#L25  

The builder reads files seen as imports (transitively) looking for the classes and documentation to display in the gallery. The search is depth first so it turns out that we were just getting lucky before and finding what was needed before reaching the import that doesn't exist yet.

By changing to a single import, the files that actually contain the classes and docs the builder is looking for will not be read until much later.

I think one possible solution is to just avoid searching in .template.dart files because they should never contain what the builder is looking for.

**Comment by natebosch** [comment]
What about checking `canRead` first and ignoring anything which isn't generated yet?

**Comment by nshahan** [comment]
Fixed in #77.
